src/libraries/Microsoft.PowerFx.Interpreter/Marshal/TypeCoercionProvider.cs

```diff
@@ -145,5 +195,101 @@ internal static bool TryCoerceTo(this FormulaValue value, FormattingInfo formatI
         {
             return TryGetDateTime(formatInfo, IRContext.NotInSource(FormulaType.DateTime), value, out result);
         }
+
+        /// <summary>
+        /// Try to convert value to Record format.
+        /// </summary>
+        /// <param name="value">Input value.</param>
+        /// <param name="targetType">Target type.</param>
+        /// <param name="result">Result value.</param>
+        /// <returns>True/False based on whether function can convert from original type to Record type.</returns> 
+        public static bool TryCoerceTo(this RecordValue value, RecordType targetType, out RecordValue result)
+        {
+            result = null;
+            var recordResult = new NamedValue[value.Fields.Count()];
+            int i = 0;
+
+            foreach (var field in value.Fields)
+            {
+                var fieldValue = value.GetField(field.Name);
+                if (!targetType.TryGetFieldType(field.Name, out FormulaType fieldType))
+                {
+                    return false;
+                }
+
+                if (!TryCoerceToTargetType(fieldValue, fieldType, out FormulaValue fieldResult))
```

Comment:
>TryCoerceToTargetType [](http://example.com/codeflow?start=21&length=21)

need to make sure this call supports recursion.

Reply:
yes this should work for recursion